### PR TITLE
Addition of scintillator rise time simulation

### PR DESCRIPTION
**Pull request by tgrbrooks**
Simulating the rise time of a scintillator for WbLS production.
Corresponding material property is SCINT_RISE_TIME, if not set will default to zero rise time.
Compared photon production time with and without rise time switched on:
rise_time_on_off

Also ran this macro with and without rise time:
```
/glg4debug/glg4param omit_muon_processes  0.0 
/glg4debug/glg4param omit_hadronic_processes  0.0 

/rat/db/set DETECTOR experiment "Watchman"
/rat/db/set DETECTOR geo_file "Watchman/Watchman.geo"

/rat/db/set GEO[detector] material "wbls_3pct_gd_01pct_WM_0420"

/run/initialize
# BEGIN EVENT LOOP
/rat/proc lesssimpledaq
/rat/proc fitcentroid
/rat/proc count
/rat/procset update 200 

# Use IO.default_output_filename
/rat/proclast outroot
#END EVENT LOOP
/generator/add combo ibd:regexfill:poisson
#/generator/ibd/spectrum boulby_geo
/generator/vtx/set IBD 1 0 0 
/generator/pos/set detector+
/generator/rate/set 9.663984059177409e-05
/run/beamOn 500
```

Results without:
```
Run time:   1954.69 sec
CPU usage: user  1934.35 sec	sys  11.67 sec
Processor usage statistics
--------------------------
             Event source: 3.689 sec/event
            lesssimpledaq: 0.000 sec/event
              fitcentroid: 0.000 sec/event
                    count: 0.000 sec/event
                  outroot: 0.006 sec/event
--------------------------
Total: 3.695 sec/event

CountProc: Total # of events 500 (1221 triggered events)
```

Results with:
```
Run time:   1943.20 sec
CPU usage: user  1918.27 sec  sys  11.79 sec
Processor usage statistics
--------------------------
             Event source: 3.670 sec/event                                                                                                                    
            lesssimpledaq: 0.000 sec/event
              fitcentroid: 0.000 sec/event
                    count: 0.000 sec/event
                  outroot: 0.006 sec/event
--------------------------
Total: 3.676 sec/event

CountProc: Total # of events 500 (1229 triggered events)
```

Also couldn't see any obvious issues browsing the trees.

### CODE REVIEW

**Comment by MorganAskins** [comment]
Wouldn't actually changing the PDF / CDF at this line to include the rise time  avoid the MC integration and random sampling?
https://github.com/AIT-WATCHMAN/rat-pac/pull/95/files#diff-6ff9383622a530d6af8fcbbe745fced1R908

Entry::Build is called once instead of for each track.

**Comment by tgrbrooks** [comment]
Yeah that would be more efficiency and easy to implement, I'll make the changes.
The code allows for the decay times to be input as either lists of decay times and corresponding strengths or as time series data, is it safe to assume that the rise time would only need to be implemented when lists of decay times are used? 
(I suppose if time series data is used the rise time would be included in whatever calculation was used to generate that data if it was wanted?)

**Comment by MorganAskins** [comment]
Brilliant, this looks great. I was thrown for a loop until I realized that the decay constants are defined as negative numbers (really weird!). One last thing if you get the chance, could you add a warning to the user that if the rise time matches any of the decay times then you will end up with a divide by zero on that cdf?

**Comment by tgrbrooks** [comment]
Good catch!
Do you think showing a warning is enough or should I also explicitly skip that contribution to the CDF or throw an error?

**Comment by MorganAskins** [comment]
Warning should be fine.

File: src/physics/src/GLG4Scint.cc [hunk 1]
```diff
@@ -880,9 +887,16 @@ void GLG4Scint::MyPhysicsTable::Entry::Build(
       for (unsigned int j=0; j < theWaveForm->GetVectorLength(); j++) {
         G4double ampl = (*theWaveForm)[j];
         G4double decy = theWaveForm->Energy(j);
+        if ( decy == -rise_time ){
+          G4cout << "\nWarning! Scintillator decay time equal to rise time.";
+          G4cout << "\nThis will give undefined scintillation production times due to divide by zero error.";
```
Comment (MorganAskins) [inline src/physics/src/GLG4Scint.cc:892]: Want to throw an endl on that last statement real quick?

**Pull request merged**